ui/page/main_page.go

```diff
@@ -554,31 +606,47 @@ func (mp *MainPage) Layout(gtx layout.Context) layout.Dimensions {
 }
 
 func (mp *MainPage) LayoutUSDBalance(gtx layout.Context) layout.Dimensions {
-	return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
-		layout.Rigid(func(gtx C) D {
-			if mp.usdExchangeSet && mp.dcrUsdtBittrex.LastTradeRate != "" && len(mp.totalBalanceUSD) > 0 {
-				inset := layout.Inset{
-					Top:  values.MarginPadding3,
-					Left: values.MarginPadding8,
-				}
-				border := widget.Border{Color: mp.Theme.Color.Gray2, CornerRadius: unit.Dp(8), Width: unit.Dp(0.5)}
-				return inset.Layout(gtx, func(gtx C) D {
-					padding := layout.Inset{
-						Top:    values.MarginPadding3,
-						Bottom: values.MarginPadding3,
-						Left:   values.MarginPadding6,
-						Right:  values.MarginPadding6,
-					}
-					return border.Layout(gtx, func(gtx C) D {
-						return padding.Layout(gtx, func(gtx C) D {
-							return mp.Theme.Body2(mp.totalBalanceUSD).Layout(gtx)
-						})
-					})
-				})
+	switch {
+	case mp.usdExchangeSet && mp.isLoadingUSDValue && mp.dcrUsdtBittrex.LastTradeRate == "":
+		gtx.Constraints.Max.Y = gtx.Px(values.MarginPadding18)
+		gtx.Constraints.Max.X = gtx.Constraints.Max.Y
+		return layout.Inset{
+			Top:  values.MarginPadding8,
+			Left: values.MarginPadding5,
+		}.Layout(gtx, func(gtx C) D {
+			return mp.loader.Layout(gtx)
```

Comment:
Instead of having a `mp.loader` property always assigned and ready, which may never be used or will not be needed after some point, create the loader object here and lay it out - strictly when needed.
```suggestion
			loader := material.NewTheme(gofont.Collection()).Loader(mp.Theme)
			return loader.Layout(gtx)
```